app/src/main/java/com/github/factotum_sdp/factotum/ui/directory/ContactCreation.kt

```diff
@@ -5,44 +5,115 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.ArrayAdapter
+import android.widget.Button
+import android.widget.EditText
 import android.widget.Spinner
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.findNavController
+import com.github.factotum_sdp.factotum.MainActivity
 import com.github.factotum_sdp.factotum.R
+import com.github.factotum_sdp.factotum.placeholder.Contact
 
 /**
  * A simple ContactCreation fragment.
  */
 class ContactCreation : Fragment() {
 
     // Should not stay like that and instead roles should use roles from future ENUM
-    val ROLES = listOf("Boss", "Courier", "Client")
+    private val roles = listOf("Boss", "Courier", "Client")
+    private var currentContact: Contact? = null
+    private val isUpdate: Boolean
+        get() = currentContact != null
+    private lateinit var viewModel : ContactsViewModel
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-    }
+    private lateinit var name: EditText
+    private lateinit var surname: EditText
+    private lateinit var address: EditText
+    private lateinit var phoneNumber: EditText
+    private lateinit var details: EditText
 
-    override fun onCreateView(
-        inflater: LayoutInflater, container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ): View? {
-        // Inflate the layout for this fragment
+    private lateinit var spinner: Spinner
 
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        // Inflate the layout for this fragment
         return inflater.inflate(R.layout.fragment_contact_creation, container, false)
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
```

Comment:
Encapsulate the logic in the onCreateView function into smaller, more focused functions. This would make it easier to understand what each piece of code is doing and make it easier to test each function separately.

Reply:
Good point.